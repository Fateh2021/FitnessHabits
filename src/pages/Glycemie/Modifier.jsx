import firebaseDb from "../../firebaseConfig";
import React, { useState, useEffect } from 'react';
import '../Tab1.css'
import {
    IonCard,
    IonButton,
    IonCardContent,
    IonDatetime,
    IonCardHeader,
    IonCardTitle,
    IonPage,
    IonSelect,
    IonSelectOption
} from "@ionic/react";


const Modifier = (props) => {

    const initialFieldValues = {
        id: '',
        jour: '',
        heure: '',
        notifierAvant: ''
    }

    var [values, setValues] = useState(initialFieldValues)

    // populate fields
    useEffect(() => {
        if (props.currentId == '')
            setValues({
                ...initialFieldValues
            })
        else
            setValues({
                ...props.glycemie[props.currentId]
            })
    }, [props.currentId, props.glycemie])

    // edit db
    const edit = obj => {
        firebaseDb.child(`NotificationGlycemie/${props.currentId}`).set(
            obj,
            err => {
                if (err)
                    console.log(err);
                else
                    window.location.href = "/glycemie";
            }
        )
    }

    const handleInputChange = e => {
        var { name, value } = e.target
        setValues({
            ...values,
            [name]: value
        })
    }

    const handleFormSubmit = e => {
        e.preventDefault();
        edit(values)
    }

    const jour = ["Lundi", "Mardi", "Mercredi", "Jeudi", "Vendredi", "Samedi", "Dimanche"];

    return (
        <IonPage>

            <form onSubmit={handleFormSubmit}>

                <IonCard>
                    <IonCardHeader>
                        <IonCardTitle>Jour</IonCardTitle>
                    </IonCardHeader>

                    <IonCardContent>
                        <IonSelect name="jour" multiple={true} value={values.jour} onIonChange={handleInputChange}>
                            {jour.map((option, i) => (
                            <IonSelectOption value={option} key={i}>{option}</IonSelectOption>))} 
                        </IonSelect>
                    </IonCardContent>
                </IonCard>

                <IonCard>
                    <IonCardHeader>
                        <IonCardTitle>Heure</IonCardTitle>
                    </IonCardHeader>

                    <IonCardContent>
                        <IonDatetime name="heure" display-format="h:mm A" picker-format="h:mm: A" value={values.heure} onIonChange={handleInputChange}></IonDatetime>
                    </IonCardContent>
                </IonCard>

                <IonCard>
                    <IonCardHeader>
                        <IonCardTitle>Notifier avant:</IonCardTitle>
                    </IonCardHeader>

                    <IonCardContent>
                        <IonDatetime name="notifierAvant" display-format="mm" picker-format="mm" value={values.notifierAvant} onIonChange={handleInputChange}></IonDatetime>
                    </IonCardContent>
                </IonCard>

                <IonButton expand="block" type="submit">Terminer</IonButton>
            </form>

        </IonPage>
    );

};

export default Modifier;
