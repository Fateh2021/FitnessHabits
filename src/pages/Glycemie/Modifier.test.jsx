 import React from 'react';
 import renderer from 'react-test-renderer';
 import Modifier from './Modifier.jsx'
 
 
 test('Modifier', ()=> {
 const component = renderer.create(<Modifier/>);
 const tree = component.toJSON();
 expect(tree).toMatchSnapshot();
 
 });
 
 
