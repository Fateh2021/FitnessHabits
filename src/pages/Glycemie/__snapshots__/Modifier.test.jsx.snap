// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modifier 1`] = `
<div
  className="ion-page"
>
  <form
    onSubmit={[Function]}
  >
    <ion-card
      onClick={[Function]}
    >
      <ion-card-header>
        <ion-card-title>
          Jour
        </ion-card-title>
      </ion-card-header>
      <ion-card-content>
        <ion-select>
          <ion-select-option>
            Lundi
          </ion-select-option>
          <ion-select-option>
            Mardi
          </ion-select-option>
          <ion-select-option>
            Mercredi
          </ion-select-option>
          <ion-select-option>
            Jeudi
          </ion-select-option>
          <ion-select-option>
            Vendredi
          </ion-select-option>
          <ion-select-option>
            Samedi
          </ion-select-option>
          <ion-select-option>
            Dimanche
          </ion-select-option>
        </ion-select>
      </ion-card-content>
    </ion-card>
    <ion-card
      onClick={[Function]}
    >
      <ion-card-header>
        <ion-card-title>
          Heure
        </ion-card-title>
      </ion-card-header>
      <ion-card-content>
        <ion-input />
      </ion-card-content>
    </ion-card>
    <ion-card
      onClick={[Function]}
    >
      <ion-card-header>
        <ion-card-title>
          Notifier avant:
        </ion-card-title>
      </ion-card-header>
      <ion-card-content>
        <ion-input />
      </ion-card-content>
    </ion-card>
    <ion-button
      onClick={[Function]}
    >
      Terminer
    </ion-button>
  </form>
</div>
`;
